Sources/LeafKit/LeafError.swift

```diff
@@ -0,0 +1,111 @@
+/// `Error` types and helper types for reporting and analyzing errors during template processing/rendering/
+
+public struct LeafError: Error {
```

Comment:
Error types moved to their own file:
* LeafError changed from `Enum` to `Struct` wrapping cases and adding source-reporting.
* LexerError adds an internal `.recoverable` flag since some lex errors may potentially be recoverable